### PR TITLE
Hit the API

### PR DESCRIPTION
**Pull request by Talha-Hanif5666**
## Highlights
- [ ] 🟢I connected the bookstore with the API provided by microverse. ✅
- [ ] 🟢 I added some createAysncThunks here and there, with some sprinkling of Axios API.✅
- [ ] 🟢The bookstore now gets its data from the cloud ✅

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -1,55 +1,66 @@
...
+      title: book[0].title,
+      author: book[0].author,
+      category: book[0].category,
+    });
+  });
+  dispatch({
+    type: GET_BOOKS,
+    payload: allBook,
+  });
+};
```
Comment (Owusu-Desmond) [inline src/redux/books/bookSlice.js]: - [ ] Great working on the functionality (adding and removing books) of the application :+1:  but from the project requirement, You should use [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls so please make sure that you refactor your code and use createAsyncThunk to make API calls. :pray:

File: src/redux/books/bookSlice.js [hunk 2]
```diff
@@ -1,55 +1,66 @@
...
 
-const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  payload: {
+const removeBook = (id) => async (dispatch) => {
+  await axios.delete(`${api}/${id}`);
+  dispatch({
+    type: REMOVE_BOOK,
     id,
```
Comment (Owusu-Desmond) [inline src/redux/books/bookSlice.js]: - [ ] Great working on the functionality (adding and removing books) of the application :+1:  but from the project requirement, You should use [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls so please make sure that you refactor your code and use createAsyncThunk to make API calls. :pray:

File: src/redux/books/bookSlice.js [hunk 3]
```diff
@@ -1,55 +1,66 @@
-import { v4 as uuidv4 } from 'uuid';
+/* eslint-disable camelcase */
+import axios from 'axios';
```
Comment (Owusu-Desmond) [inline src/redux/books/bookSlice.js]: - [ ] Please make sure that  [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) is imported in your  project to make API calls :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @Talha-Hanif5666 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — APPROVED** [review]
Hi @Talha-Hanif5666, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- createAsyncThunk is used 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**